Dockerfile

```diff
@@ -6,20 +6,53 @@ RUN mkdir -p /tmp && chown -R www-data:www-data /tmp && chmod -R 777 /tmp
 RUN mkdir -p /var/run/ && chown -R www-data:www-data /var/run && chmod -R 777 /var/run
 USER www-data
 
-FROM base AS prod
+# We install the dependencies in a separate layer as the frontend image also needs them
+FROM base AS vendor
 USER root
-COPY . /var/www/html
+
+# Copying the full context is not great for caching
+# But composer install executes symfony commands that need the full context
+COPY . /var/www/html/
 
 RUN composer install
```

Comment:
You can do a composer install without script to avoid passing the full context (just the composer.lock/json)
> composer install --no-interaction  --no-progress --no-scripts --optimize-autoloader

You will need to run it a second time with the script. but you will win some time packing the vendors. (also I am not sure if you need to use `--no-plugins`)

Reply:
Also add --no-dev

Reply:
> You will need to run it a second time with the script. but you will win some time packing the vendors. (also I am not sure if you need to use --no-plugins)

Indeed, makes sense.
I thought of doing it but didn't want to complexify the build too much.

> Also add --no-dev

Last time I tried this it failed as symfony absolutely wanted some dev-specific stuff to be present with the prod env. I will try that again and if it works I'll keep it.

Reply:
Applied.
And I discovered that I just needed to add an `APP_ENV=prod` before the composer install to run the script, but as stated along with that comment, it then fails accessing the DB